python/pyarrow/tests/test_compute.py

```diff
@@ -1427,9 +1428,20 @@ def test_extract_datetime_components():
                   "2008-12-28",
                   "2008-12-29",
                   "2012-01-01 01:02:03"]
+    timezones = ["UTC", "US/Central", "Asia/Kolkata",
+                 "Etc/GMT-4", "Etc/GMT+4", "Australia/Broken_Hill"]
 
+    # Test timezone naive timestamp array
     _check_datetime_components(timestamps)
 
+    # Test timezone aware timestamp array
+    if sys.platform == 'win32':
+        # TODO: We should test on windows once ARROW-13168 is resolved.
+        pytest.skip('Timezone database is not available on Windows yet')
+    else:
+        for timezone in timezones:
+            _check_datetime_components(timestamps, timezone)
```

Comment:
@rok seems like the timezone argument is not used within the `_check_datetime_components` helper, so I think it doesn't have any effect.

Reply:
Oh, indeed. Let me fix that.

Reply:
Done.

Reply:
@rok I executed the same test case (copied the `test_compute.py` file) on top of the master branch and the test case passes. If I understand correctly this should fail on master since it is a new feature.

Reply:
Right, it should trip [this check](https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/scalar_temporal.cc#L81).
Let me check some more.

Reply:
Timezone is now passed properly but a weekday issue appeared. Let me see.

Reply:
It turned out that `DayOfWeekExec` was not passing the timezone to the `DayOfWeek` kernel. Should be fixed now.